src/utils/transaction/insertTransaction.ts

```diff
@@ -58,98 +152,70 @@ export const insertTransaction = async (
     target: getChecksumAddress(insertedTransaction.target),
     sender: getChecksumAddress(insertedTransaction.sender),
     value: insertedTransaction.value ?? 0n,
+    walletType: "local", // we override this later
   };
 
+  // First attempt: try to find wallet by 'from' address
   let walletDetails: ParsedWalletDetails | undefined;
+  let walletFoundByFrom = false;
 
   try {
     walletDetails = await getWalletDetails({
-      address: queuedTransaction.from,
+      walletAddress: queuedTransaction.from,
     });
+    walletFoundByFrom = true;
+  } catch {}
 
-    // when using the v5 SDK with smart backend wallets, the following values are not set correctly:
-    // isUserOp is set to false
-    // account address is blank or the user provided value (this should be the SBW account address)
-    // from is set to the SBW account address (this should be the SBW signer address)
-    // these values need to be corrected so the worker can process the transaction
+  // Case 1 & 2: Wallet found by 'from'
+  if (walletFoundByFrom && walletDetails) {
     if (isSmartBackendWallet(walletDetails)) {
-      if (queuedTransaction.accountAddress) {
-        throw createCustomError(
-          "Smart backend wallets do not support interacting with other smart accounts",
-          StatusCodes.BAD_REQUEST,
-          "INVALID_SMART_BACKEND_WALLET_INTERACTION",
-        );
-      }
-
-      if (
-        !(await doesChainSupportService(
-          queuedTransaction.chainId,
-          "account-abstraction",
-        ))
-      ) {
-        throw createCustomError(
-          "Chain does not support smart backend wallets",
-          StatusCodes.BAD_REQUEST,
-          "SBW_CHAIN_NOT_SUPPORTED",
-        );
-      }
-
-      queuedTransaction = {
-        ...queuedTransaction,
-        isUserOp: true,
-        signerAddress: walletDetails.accountSignerAddress,
-        from: walletDetails.accountSignerAddress,
-        accountAddress: queuedTransaction.from,
-        target: queuedTransaction.to,
-        accountFactoryAddress: walletDetails.accountFactoryAddress ?? undefined,
-        entrypointAddress: walletDetails.entrypointAddress ?? undefined,
-      };
+      // Case 1: V5 Smart Backend Wallet
+      queuedTransaction = await transformV5SmartBackendWallet(
+        queuedTransaction,
+        walletDetails,
+      );
+    } else {
+      // Case 2: Regular wallet (V4 or V5) - just add wallet type
+      queuedTransaction.walletType = walletDetails.type;
+    }
+  } else {
+    // From this point on, we're in Case 3 territory - check for V4 smart backend wallet
+    if (!queuedTransaction.accountAddress) {
+      throw createCustomError(
+        "Account not found",
```

Comment:
Is it possible for a user to get into this state? If so, we should make the error more clear.
> Invalid backend wallet configuration. See: https://portal.thirdweb.com/engine/troubleshooting

And maybe put more help there.

If a user can't get here and this will only fail if there's a bug within Engine, this should just throw (treated as a 500).

Reply:
Same with the next two thrown errors below.